include/nlohmann/detail/iterators/iteration_proxy.hpp

```diff
@@ -3,104 +3,105 @@
 #include <cstddef> // size_t
 #include <string> // string, to_string
 #include <iterator> // input_iterator_tag
+#include <tuple> // tuple_size, get, tuple_element
 
 #include <nlohmann/detail/value_t.hpp>
+#include <nlohmann/detail/meta/type_traits.hpp>
 
 namespace nlohmann
 {
 namespace detail
 {
-/// proxy class for the items() function
-template<typename IteratorType> class iteration_proxy
+template <typename IteratorType> class iteration_proxy_internal
```

Comment:
Discussion: Maybe rename this to `iteration_proxy_value`?

Reply:
Yes.
That sounds like a nice enough name and does provide information regarding what `iteration_proxy_internal `exactly does.

Reply:
I'm ok with `iteration_proxy_value`

Reply:
I also prefer `iteration_proxy_value` over `iteration_proxy_kv` or `iteration_proxy_key_value`.

Reply:
@nlohmann I corrected that and updated the branch with another Squash of the commits.
Could you please check and see if they are proper?
One of my many doubts is [here](https://github.com/nlohmann/json/pull/1391#discussion_r242005973)

Reply:
I just added an answer.